### DISCUSSION TITLE
The shortcode "hextra/hero-badge" not found

### DISCUSSION THREAD

**Discussion by confusedkernel** [imfing/hextra#252] [Q&A]
Hello! I'm using hextra's shortcode hero-badge on my index.html, however when I deploy it to GitHub page it couldn't build because it couldn't find the template for shortcode "hextra/hero-badge."

The code below was the error:
```
Total in 9 ms
Error: error building site: assemble: "/home/runner/work/nottyl.github.io/nottyl.github.io/content/_index.md:9:6": failed to extract shortcode: template for shortcode "hextra/hero-badge" not found
Error: Process completed with exit code 1.
```

However, this error still shows up even after I commented out the section with hero-badge shortcode. Does anyone know why this happens? And how do I fix it? Thanks in advance.

**Comment by imfing** [comment]
@nottyl Hi, I haven't seen such issue before. Could you provide an example repository that reproduces this issue?

How did you set up the repository? Did you use the starter template?

**Reply by WhiteIce39** [reply]
Sorry for disturbing you, but I am facing similar problem.
Firstly, I initialized a new Hugo site and added Hextra theme as a Git submodule in my PC, and then I previewed the site locally, it turn out well. However, After I push the local files onto my Github repository, when I try to use Github Pages with Github Action to publish it, it obstructed. The error message like this:

Error: error building site: process: readAndProcessContent: "/home/runner/work/whiteice39-blog/whiteice39-blog/content/_index.md:6:1": failed to extract shortcode: template for shortcode "hextra/hero-badge" not found

Sorry for my poor ability, I can't figure out what is the problem. The URL of the repository I set up is "github.com/WhiteIce39/whiteice39-blog".
 
I am a green hand tring to set up a website, sincerely appreciate your help.

Well, I might figure out the reason, the themes/hextra/ directory I uploaded is a link to another repository, which as a submodule, causing Github pages can not recogonize the files.

But there is a new issue, part of the new error message is :

execute of template failed: template: _shortcodes/card.html:23:24: executing "_shortcodes/card.html" at <.Process>: error calling Process: this feature is not available in your current Hugo version

There is still a long way to go for me to learn.

Anyway, I have solved all the errors for now......the website works well now.
